core/src/main.cpp

```diff
@@ -946,11 +948,17 @@ auto main(int _argc, char* _argv[]) -> int
 		}
 
 		if (vm.count("config-file") == 0) {
-			fmt::print(stderr, "Error: Missing [CONFIG_FILE_PATH] parameter.");
+			fmt::print(stderr, "Error: Missing [CONFIG_FILE_PATH] parameter.\n");
 			return 1;
 		}
 
-		const auto config = json::parse(std::ifstream{vm["config-file"].as<std::string>()});
+		const auto& config_file_path = vm["config-file"].as<std::string>();
+		if (!fs::is_regular_file(config_file_path)) {
```

Comment:
is_regular_file will check if it can be opened?   considering it exists, but has bad permissions...

Reply:
No. It confirms the file is NOT something other than `regular`. See https://en.cppreference.com/w/cpp/filesystem/file_type for the various file types that can be returned.

Hmm, but now I'm thinking it should just be an existence check since it distinguishes symlinks.

Reply:
so if the file exists, and is regular, but cannot be read because of permissions...

does this catch that?  or we handle that later when it tries to read/parse it?

Reply:
Confirmed the changes do not cover permissions. If the application cannot read the file, you get the same error that was reported in the issue.

How common is it for applications to check permissions for config files though? Don't think I've ever encountered a situation with that before.

Reply:
ah, right... it will just say, cannot access... which is correct.    okay, so an existence check is probably sufficient here.

Reply:
Updated code to report an error if the file does not exist or cannot be opened.

Done.